core/src/main/java/org/infinispan/distribution/ch/AffinityTaggedKey.java

```diff
@@ -0,0 +1,12 @@
+package org.infinispan.distribution.ch;
+
+public interface AffinityTaggedKey {
+
+   /**
+    * This numeric id is used exclusively for storage affinity in Infinispan.
+    *
+    * @return the segment id to be used for storage, or -1 to fall back to normal owner selection.
```

Comment:
> Maybe we could throw something to indicate that we need to get back to normal owner? 

Throwing an exception rather than returning -1 is considerably slower, apart for polluting every caller code, I personally prefer to avoid it. Maybe it'd possible to get rid of this ```-1``` altogether and assume that this key will always have a valid segment, will check.

> Or maybe some isEnabled flag?

Could you elaborate on that? Would it be set via configuration?